ci/infra/openstack/generate-cloud-provider-conf

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+
+if [ -z "${OS_AUTH_URL}" ] || [ -z "${OS_USERNAME}" ] || \
+   [ -z "${OS_PASSWORD}" ] || [ -z "${OS_PROJECT_ID}" ] || \
+   [ -z "${OS_PRIVATE_SUBNET_ID}" ] || [ -z "${OS_PUBLIC_NET_ID}" ]; then
```

Comment:
Maybe consider `[[` for at least this test? With the double bracket keyword, not only do variables not need quoted, but the "or" can also go in the test. Getting rid of a lot of extra punctuation would make this quite a bit easier on the eyes, IMO. The double bracket is supported on any bash that this will ever run under.

But this is legit olde tyme shell style that still works on posix /bin/sh, so that's fine if you'd prefer to keep it this way for backwards compatibility. It's way less efficient, since single bracket is a built-in vs a keyword, but it's not like performance is the main goal here.

Reply:
let's keep it